### PR TITLE
zoneminder: 1.32.3 -> 1.34.3

### LINKED ISSUE #54630: Zoneminder auto-upgrades are not supported
## Issue description
https://github.com/NixOS/nixpkgs/pull/53495 added Zoneminder, but it seems there is no code that handles upgrades, while that is provided by upstream and for example the Ubuntu packaging.

It seems to be more of a developer preview version on NixOS than something that could be used by end-users in a production environment. I would have added a flag like `service.zoneminder.i_am_a_developer_and_know_this_breaks_on_upgrades` that needs to be set to true before it does something for that reason.   
 
I welcome the progress that has been made, but it doesn't appear to be in good enough shape to compete with other solutions. 

I will open another issue for adding declarative camera support (https://github.com/NixOS/nixpkgs/issues/54631), because there was another user who also wanted to have that, which should also make this issue smaller.

### PR DESCRIPTION
**Pull request by danielfullmer**
###### Motivation for this change
Update to latest version. The version currently in nixpkgs has a lot of vulnerabilities.
See: https://github.com/ZoneMinder/zoneminder/releases

A user with an existing database would also need to run `zmupdate.pl` as the `zoneminder` user. I've added some missing perl runtime dependencies for `zmupdate.pl` as well.

I've tested this with my currently running zoneminder instance and it seems to be running correctly after the upgrade.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

CC @peterhoeg 
Closes: #54630

### CODE REVIEW

**Review by veprbl — DISMISSED** [review]
LGTM

**Comment by aanderse** [comment]
@danielfullmer can we run `zmupdate.pl` on every service start so the user doesn't have to? Or do we add notes somewhere mentioning this?

**Comment by danielfullmer** [comment]
@aanderse It looks like we can run `zmupdate.pl` on startup. I originally thought it might be a problem since the script is interactive by default--but the `-nointeractive` flag works for our use. The script doesn't do anything if the database is already at the current version.

Just added a commit addressing this. I've tested upgrading from `1.32.3` to `1.34.2` in a VM and it seems to work fine.

**Comment by veprbl** [comment]
I had my reservations about forcing an automatic schema update back when discussed in #54630. However, I'm not using zoneminder on NixOS, so I, personally, don't mind the change.

**Comment by aanderse** [comment]
@danielfullmer sorry my bad. I forgot that automatic database schema updates aren't universally accepted among the team. This is a judgement call that you can make. Keep the schema update or not. I think I'll start a discourse thread to get some feedback on the topic in general.

**Comment by danielfullmer** [comment]
Any commiters please hold off on merging this for the time being. I've found an issue with the UI being unresponsive when configuring cameras that I would need to debug.  (or anyone else interested)

With regard to automatic updating--I'm open to either option. If we do go for automatic schema updates I'd love to have someone else test it as well before this gets merged.  One nice feature of manually running `zmupdate.pl` interactively is that it has optional functionality to make a database backup before updating.

I'm also very busy with my thesis at the moment so I'm not sure if I will get to fixing the UI issue in the next few days.

**Comment by danielfullmer** [comment]
I've updated this PR with a few changes. It should be ready to merge after review.

Updated to [1.34.3](https://github.com/ZoneMinder/zoneminder/releases/tag/1.34.3).
Added a fix for improper timezone detection.

As far as automatic schema updates. I think we should do this only if `services.zoneminder.database.createLocally` is enabled. Zoneminder can be used with multiple instances backed by the same remote database--and we almost surely do not want to do automatic updates in that case. Conditionally updating the schema if it is a local database is also what upstream does in their [debian postinstall script](https://github.com/ZoneMinder/zoneminder/blob/c29f2cb8cfec536e444af85f71fde66ed4e58c82/distros/debian/postinst).

The UI issue was a result of zoneminder's "cache busting". They create symlinks in `/var/cache/zoneminder` to the js/css files, where the symlink has the timestamp of the destination file in the filename. This was intended to ensure that newer files would get a different URL. However, in the nix store, the files all have the timestamp "1". So when I upgraded, these symlinks would still point to the old version of zoneminder, and were not replaced. My solution was to patch the symlink filename calculation to replace the timestamp with just the hash of the zoneminder source.

File: pkgs/servers/zoneminder/default.nix
Comment (veprbl) [inline pkgs/servers/zoneminder/default.nix]: I guess, it makes more sense to tag cache with ``` `basename $out` ``` instead of just `$src`.

Comment (danielfullmer) [reply pkgs/servers/zoneminder/default.nix]: OK. I've tested and pushed a new commit for this change.

**Review by veprbl — APPROVED** [review]
Looks reasonable, but I can't test this now.

**Comment by veprbl** [comment]
@danielfullmer Please squash the last two commits and edit "zoneminder: 1.32.3 -> zoneminder 1.34.3" to "zoneminder: 1.32.3 -> 1.34.3"

**Comment by danielfullmer** [comment]
@veprbl Just squashed those commits and fixed the commit title

**Pull request merged**
